src/stlink-lib/chipid.c

```diff
@@ -55,29 +55,23 @@ void process_chipfile(char *fname) {
 
     ts = calloc(sizeof(struct stlink_chipid_params), 1);
 
-    while (fgets(buf, 1024, fp) != NULL) {
-        for (p = buf; isspace (*p); p++);
+    while (fgets(buf, sizeof(buf), fp) != NULL) {
 
-        if (!*p) {
-            continue; // we hit end-of-line with only whitespace
-        }
-
-        if (*p == '#') {
+        if(strncmp(buf, "#", strlen("#")) == 0)
             continue; // ignore comments
```

Comment:
I don’t like the redundant braces, but let’s follow the existing style:

```suggestion
        if (strncmp(buf, "#", strlen("#")) == 0) {
            continue; // ignore comments
        }
```